src/main/java/software/amazon/awssdk/crt/http/HttpStreamManager.java

```diff
@@ -0,0 +1,213 @@
+        return new HttpStreamManager(options);
+    }
+
+    private HttpClientConnectionManagerOptions getConnManagerOptFromStreamManagerOpt(HttpStreamManagerOptions options) {
+
+        HttpClientConnectionManagerOptions connManagerOptions = new HttpClientConnectionManagerOptions();
+        connManagerOptions.withClientBootstrap(options.getClientBootstrap())
+                .withSocketOptions(options.getSocketOptions())
+                .withTlsContext(options.getTlsContext())
+                .withWindowSize(options.getWindowSize())
+                .withUri(options.getUri())
+                .withMaxConnections(options.getMaxConnections())
+                .withPort(options.getPort())
+                .withProxyOptions(options.getProxyOptions())
+                .withManualWindowManagement(options.isManualWindowManagement())
+                .withMonitoringOptions(options.getMonitoringOptions());
+        return connManagerOptions;
+    }
+
+    private HttpStreamManager(HttpStreamManagerOptions options) {
+        HttpClientConnectionManagerOptions connManagerOptions = getConnManagerOptFromStreamManagerOpt(options);
+        this.connectionManager = HttpClientConnectionManager.create(connManagerOptions);
+        this.h2StreamManager = Http2StreamManager.create(options);
+        this.shutdownComplete = new CompletableFuture<Void>();
+        this.connectionManager.getShutdownCompleteFuture().whenComplete((v, throwable) -> {
+            if (throwable != null) {
+                this.shutdownCompleteException = throwable;
+            }
+            long shutdownNum = this.shutdownNum.addAndGet(1);
+            if (shutdownNum == 2) {
+                /* both connectionManager and the h2StreamManager has been shutdown. */
+                if (this.shutdownCompleteException != null) {
+                    this.shutdownComplete.completeExceptionally(this.shutdownCompleteException);
+                } else {
+                    this.shutdownComplete.complete(null);
+                }
+            }
+        });
+        this.h2StreamManager.getShutdownCompleteFuture().whenComplete((v, throwable) -> {
+            if (throwable != null) {
+                this.shutdownCompleteException = throwable;
+            }
+            long shutdownNum = this.shutdownNum.addAndGet(1);
+            if (shutdownNum == 2) {
+                /* both connectionManager and the h2StreamManager has been shutdown. */
+                if (this.shutdownCompleteException != null) {
+                    this.shutdownComplete.completeExceptionally(this.shutdownCompleteException);
+                } else {
+                    this.shutdownComplete.complete(null);
+                }
+            }
+        });
+    }
+
+    /**
+     * Get the protocol version the manager runs on.
+     */
+    public HttpVersion getHttpVersion() throws Exception {
+        if (this.h2StreamManager != null && this.connectionManager != null) {
```

Comment:
Let's think about the case if server have two different host that host both http/1.1 and http/2. And, they give different protocol overtime. How will we handle it? Error out? Or probably handle it someway.